src/test/java/uk/nhs/digital/nhsconnect/lab/results/translator/mapper/ObservationMapperTest.java

```diff
@@ -484,7 +484,7 @@ void testMapTestGroupsWithBidirectionalReferences() {
         final var uuidToUrlMap = Map.of("uuid-1", "url-1", "uuid-2", "url-2");
         when(uuidGenerator.generateUUID()).thenReturn("uuid-1", "uuid-2");
         uuidToUrlMap.forEach((uuid, url) ->
-            when(fullUrlGenerator.generate(uuid)).thenReturn(url));
+            lenient().when(fullUrlGenerator.generate(uuid)).thenReturn(url));
```

Comment:
Why this change? Is this line even still needed?

Reply:
It is needed but only for some value. I just didn't want to spend more time on figuring this out

Reply:
Ah I see, in some cases it's hit and some cases it's not